src/lib/libc_preload/libc_impls.c

```diff
@@ -33,10 +33,21 @@
 // This function drives all of our wrappers over to the shim.
 long syscall(long n, ...) {
     va_list(args);
+
+    // We used to provide shim_api_syscallv that took a va_list, but it's not
+    // clear how to wire such an API through Rust.  just enumerate the arguments
+    // and call shim_api_syscall instead.
+    
     va_start(args, n);
-    long rv = shim_api_syscallv(n, args);
+    long arg1 = va_arg(args, long);
+    long arg2 = va_arg(args, long);
+    long arg3 = va_arg(args, long);
+    long arg4 = va_arg(args, long);
+    long arg5 = va_arg(args, long);
+    long arg6 = va_arg(args, long);
     va_end(args);
-    return rv;
+
+    return shim_api_syscall(n, arg1, arg2, arg3, arg4, arg5, arg6);
 }
```

Comment:
Won't this read uninitialized/invalid memory if they passed fewer than 6 arguments?

Reply:
These arguments are guaranteed to be passed in registers by the ABI. `va_start` will have initialized them with *some* register values, even if the caller didn't explicitly pass them.

I agree it feels a bit sketchy, but is also what e.g. glibc does and is basically the only way to implement `syscall` without mapping every syscall number to number of expected arguments. https://github.com/bminor/glibc/blob/d2fda60e7c4072180ba91df46bbbdacc0f4a133c/sysdeps/unix/sysv/linux/syscall.c#L23